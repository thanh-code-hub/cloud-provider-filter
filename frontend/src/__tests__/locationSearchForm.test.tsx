import { fireEvent, waitFor } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import LocationSearchForm from '../components/locationSearchForm';

describe('LocationSearchForm', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly', () => {
    const { container } = render(<LocationSearchForm onLocationFound={() => {}} disabled={false} />);
    expect(container).toMatchSnapshot();
  });

  test('test failing location search', async () => {
    jest.spyOn(global, 'fetch').mockImplementationOnce(
      jest.fn(() =>
        Promise.resolve({
          ok: true,
          status: 200,
          json: () => Promise.resolve([]),
        }),
      ) as jest.Mock,
    );

    render(<LocationSearchForm onLocationFound={() => {}} disabled={false} />);

    fireEvent.change(screen.getByRole('location-search-input'), { target: { value: 'zzpzzpzpzpzp' } });
    fireEvent.click(screen.getByRole('search-button'));

    await waitFor(() => {
      expect(screen.getByText('No location found')).toBeInTheDocument();
    });
  });
});
