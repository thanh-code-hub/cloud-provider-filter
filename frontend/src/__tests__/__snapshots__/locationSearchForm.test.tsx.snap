// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationSearchForm renders correctly 1`] = `
<div>
  <div>
    <form
      role="filter-form"
    >
      <fieldset
        style="padding: 0px; margin: 0px;"
      >
        <input
          placeholder="Enter provider"
          role="provider-search-input"
          type="text"
          value=""
        />
        <input
          placeholder="Enter location or address"
          role="location-search-input"
          type="text"
          value=""
        />
        <button
          role="search-button"
          type="submit"
        >
          Search cloud provider
        </button>
        <button
          role="clear-button"
        >
          Clear
        </button>
      </fieldset>
    </form>
  </div>
</div>
`;
